sdk/src/main/scala/io/horizen/account/state/ProxyMsgProcessor.scala

```diff
@@ -0,0 +1,129 @@
+package io.horizen.account.state
+
+import io.horizen.account.abi.ABIUtil.{METHOD_ID_LENGTH, getABIMethodId, getArgumentsFromData, getFunctionSignature}
+import io.horizen.account.state.ProxyMsgProcessor._
+import io.horizen.account.state.events.ProxyInvocation
+import io.horizen.account.utils.WellKnownAddresses.PROXY_SMART_CONTRACT_ADDRESS
+import io.horizen.evm.Address
+import io.horizen.params.{MainNetParams, NetworkParams}
+import io.horizen.utils.BytesUtils
+import org.web3j.utils.Numeric
+import sparkz.crypto.hash.Keccak256
+
+trait ProxyProvider {
+
+}
+
+case class ProxyMsgProcessor(params: NetworkParams) extends NativeSmartContractMsgProcessor with ProxyProvider {
+
+  override val contractAddress: Address = PROXY_SMART_CONTRACT_ADDRESS
+  override val contractCode: Array[Byte] = Keccak256.hash("ProxySmartContractCode")
+
+  // TODO this must be updated when merging with 0.8.0, we should support consensusEpochNumber for hard fork activation
+  override def init(view: BaseAccountStateView): Unit = {
+    super.init(view)
+  }
+
+  def doInvokeSmartContractStaticCallCmd(invocation: Invocation, view: BaseAccountStateView, context: ExecutionContext): Array[Byte] = {
+    doInvokeSmartContractCmd(invocation, view, context, readOnly = true)
+  }
+
+  def doInvokeSmartContractCallCmd(invocation: Invocation, view: BaseAccountStateView, context: ExecutionContext): Array[Byte] = {
+    doInvokeSmartContractCmd(invocation, view, context, readOnly = false)
+  }
+
+  private def doInvokeSmartContractCmd(invocation: Invocation, view: BaseAccountStateView, context: ExecutionContext, readOnly : Boolean): Array[Byte] = {
+
+    val msg = context.msg
+    log.debug(s"Entering with invocation: $invocation")
+    // check that message contains a nonce, in the context of RPC calls the nonce might be missing
+    if (msg.getNonce == null) {
```

Comment:
Is this check necessary? It causes rpc calls that don't provide nonce to fail.

Reply:
Handled in https://github.com/HorizenOfficial/Sidechains-SDK/pull/911